### PR TITLE
Remove Channel Groups APIs and related functions

### PR DESCRIPTION
**Pull request by jrainville**
For https://github.com/status-im/status-desktop/issues/12595

We are getting rid of the concept of channel groups. See the issue for the rationale.

This PR removes the functions and helpers for it.

I don't think mobile uses those APIs, but I might have missed something. cc @cammellos 

It will still need to be tested on both Desktop and Mobile. cc @pavloburykh

### CODE REVIEW

File: protocol/communities/community.go
Comment (igor-sirotin) [inline protocol/communities/community.go:1]: This file changes seem unrelated, maybe committed here by mistake? 
But I'm ok with with the improvements 🙂

**Review by osmaczko — APPROVED** [review]
https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExc2dqM3c0N2YyMGVvNW9tYWo5Zjlpd2xpa2o4Yzl6ZHhkdmZmZWR1ayZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/j0vs5H7Kcz3Pm9LRDa/giphy.gif

File: protocol/communities/community.go [hunk 1]
```diff
@@ -182,12 +183,17 @@ func (o *Community) MarshalPublicAPIJSON() ([]byte, error) {
...
 		for id, c := range o.config.CommunityDescription.Chats {
-			// NOTE: Here `CanPost` is only set for ChatMessage. But it can be different for reactions/pin/etc.
+			// NOTE: Here `CanPost` is only set for ChatMessage and Emoji reactions. But it can be different for pin/etc.
 			// Consider adding more properties to `CommunityChat` to reflect that.
 			canPost, err := o.CanPost(o.config.MemberIdentity, id, protobuf.ApplicationMetadataMessage_CHAT_MESSAGE)
 			if err != nil {
 				return nil, err
 			}
+			canPostReactions, err := o.CanPost(o.config.MemberIdentity, id, protobuf.ApplicationMetadataMessage_EMOJI_REACTION)
+			fmt.Println("Can I post reactions?? ", c.Identity.DisplayName, canPostReactions)
```
Comment (cammellos) [inline protocol/communities/community.go]: debug

File: protocol/communities/community.go [hunk 2]
```diff
@@ -332,6 +339,11 @@ func (o *Community) MarshalJSON() ([]byte, error) {
...
```
Comment (cammellos) [inline protocol/communities/community.go]: debug

**Comment by pavloburykh** [comment]
@jrainville hey! thanks for pinging. 

According to @cammellos since we do not use those APIs, testing on mobile can be skipped.

**Review by igor-sirotin — APPROVED** [review]
🎉 

delete-button

**Comment by jrainville** [comment]
> @jrainville hey! thanks for pinging.
> 
> According to @cammellos since we do not use those APIs, testing on mobile can be skipped.

I did change some return values for the group chat functions, but they still return a Chat, so I didn't think it would affect. We used to return a "super" Chat object, but I doubt mobile used the extra properties on them, even desktop didn't use them, so it's probably ok indeed.

File: protocol/communities/community.go
Comment (jrainville) [reply protocol/communities/community.go:1]: Ah, no they are needed for the desktop part. Since we no longer get `canPostReactions` from the chat API, we need them from the CommunityChat.

I should have listed that in the PR description, but I did so many changes that I forgot to list it :laughing:
